hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -190,9 +180,9 @@ public Address getPartitionOwner(int partitionId) {
 
     @Override
     public int getPartitionId(@Nonnull Data key) {
-        final int pc = getPartitionCount();
-        if (pc <= 0) {
-            return 0;
+        int pc = getPartitionCount();
+        if (pc == 0) {
+            throw new HazelcastClientOfflineException();
```

Comment:
is this a new behaviour? I mean we normally used this exception for async mode client, now it can be thrown for sync client as well, right?

Reply:
Partition count can not be zero for the sync mode. On the sync mode, we are waiting for the first connection to be established. And in that path, we are initializing the partition count with the value coming from the server with authentication. So this exception is still used only for async mode client.

Reply:
OK, but it is not easy to understand your explanation, you can put an explanation here.